### PR TITLE
feat: add verbose cli flag support

### PR DESCRIPTION
**Pull request by deepak1725**
This PR adds `verbose` flag support.

Build: `go build -o crda`

Execute:
```
./crda analyse /absolute/path/to/package.json --verbose
```

Development Run
```
go run main.go analyse /absolute/path/to/package.json --verbose
```

### CODE REVIEW

**Comment by deepak1725** [comment]
prompt
Interactive mode Prompt

File: analyses/stackanalyses/controller.go
Comment (arajkumar) [inline analyses/stackanalyses/controller.go:58]: It seems the behaviour fallbacks to summary Incase if user passes `--verbose`, but not eligible for verbose dump. Is this acceptable? I guess it is better to show the message to register with snyk and appropriate help command.

File: analyses/verbose/helper.go
Comment (arajkumar) [inline analyses/verbose/helper.go:1]: Please write tests for verbose pkg

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: analyses/stackanalyses/controller.go
Comment (arajkumar) [reply analyses/stackanalyses/controller.go:58]: @prashbnair , What is your opinion on this?

Comment (prashbnair) [reply analyses/stackanalyses/controller.go:58]: This is what Parag wants, I guess we can ask them to register to see the complete information.

Comment (deepak1725) [reply analyses/stackanalyses/controller.go:58]: Exactly

File: analyses/verbose/helper.go
Comment (deepak1725) [reply analyses/verbose/helper.go:1]: Tests are already In.

File: analyses/summary/helpers.go
Comment (deepak1725) [reply analyses/summary/helpers.go]: Ack.

File: analyses/verbose/helper.go
Comment (deepak1725) [reply analyses/verbose/helper.go]: Ack.

File: analyses/summary/helpers.go
Comment (deepak1725) [reply analyses/summary/helpers.go]: Ack.

**Review by dgpatelgit — COMMENTED** [review]
Changes look good to me, merge it once you get go from other reviewers with open comments.

**Comment by arajkumar** [comment]
## suggestions to auth command
```
To get Snyk Token, Please click https://app.snyk.io/redhat/snyk-token

Snyk Token [Press Enter to continue]: xxxx█
Successfully Registered.

	 crda-key: xxxx

 This token is confidential, Please keep it safe!
```
Seem to have some alignment issues.
1. Leading space in `This token is confidential, Please keep it safe!` shall be removed
2. Leading spaces in `	 crda-key: xxxx` shall be removed

How about changing "Snyk Token [Press Enter to continue]" to "Snyk Token [Press Enter to skip]".?


## suggestion to analyse
```
Analysing your Dependency Stack! Please wait...
Summary Report for Analyses:

Total Scanned Dependencies:68
Total Scanned Transitive Dependencies:106
Direct Vulnerable Dependencies:3
Total Vulnerabilities:8
Commonly Known Vulnerabilities:8
Vulnerabilities Unique to Synk:0
Critical Vulnerabilities:3
High Vulnerabilities:0
Medium Vulnerabilities:5
Low Vulnerabilities:0

(Powered by Snyk)

Use --verbose for detailed report.%
```
1. See a training % in my zsh, not sure it is the case for all
2. Can you add space between : and number? e.g. `Total Scanned Dependencies: 68`
3. Can this `Use --verbose for detailed report.` changed to `tip: use --verbose for detailed report.` 


## suggestions to verbose
```
➜  cli-tools git:(verbose) ./crda analyse ~/works/log-parser/requirements.txt --verbose
Analysing your Dependency Stack! Please wait...
Verbose Report for given Stack:

Scanned 68 Dependencies and 106 Transitives, Found 8 Issues

Fixable Issues:



	 Issues in Transitives:
	 ✒ anymarkup@0.8.1->pyyaml@5.3.1
	 ✘ Arbitrary Code Execution[Critical Severity] [https://snyk.io/vuln/SNYK-PYTHON-PYYAML-590151] in pyyaml@5.3.1
	 introduced by anymarkup@0.8.1


















	Upgrade cryptography@3.2.1 to cryptography@3.4.6
	 ✘ Cryptographic Issues[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-CRYPTOGRAPHY-1070544] in cryptography@3.2.1
	 introduced by cryptography@3.2.1


	 Issues in Transitives:
	 ✒ flask-cors@3.0.9->jinja2@2.11.2
	 ✘ Regular Expression Denial of Service (ReDoS)[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-JINJA2-1012994] in jinja2@2.11.2
	 introduced by flask-cors@3.0.9



	 Issues in Transitives:
	 ✒ flask@1.1.2->jinja2@2.11.2
	 ✘ Regular Expression Denial of Service (ReDoS)[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-JINJA2-1012994] in jinja2@2.11.2
	 introduced by flask@1.1.2












	Upgrade jinja2@2.11.2 to jinja2@2.11.3
	 ✘ Regular Expression Denial of Service (ReDoS)[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-JINJA2-1012994] in jinja2@2.11.2
	 introduced by jinja2@2.11.2















	Upgrade pyyaml@5.3.1 to pyyaml@5.4.1
	 ✘ Arbitrary Code Execution[Critical Severity] [https://snyk.io/vuln/SNYK-PYTHON-PYYAML-590151] in pyyaml@5.3.1
	 introduced by pyyaml@5.3.1



	 Issues in Transitives:
	 ✒ raven@6.10.0->jinja2@2.11.2
	 ✘ Regular Expression Denial of Service (ReDoS)[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-JINJA2-1012994] in jinja2@2.11.2
	 introduced by raven@6.10.0






	 Issues in Transitives:
	 ✒ selinon@1.0.0->pyyaml@5.3.1
	 ✘ Arbitrary Code Execution[Critical Severity] [https://snyk.io/vuln/SNYK-PYTHON-PYYAML-590151] in pyyaml@5.3.1
	 introduced by selinon@1.0.0
















(Powered by Snyk)
```
1. See lots of empty spaces, not sure it is specific to my shell(zsh with oh-my-zsh)

^^ @deepak1725

**Comment by arajkumar** [comment]
```
➜  cli-tools git:(verbose) ./crda analyse ~/works/log-parser/requirements.txt
3:03PM FTL Please run `crda auth` command first to get `crda-key` and set it in environment.
```

I still see log statements with time and log level to show errors to cli users, IMO it should be replaced with normal error print(probably with emoji).

e.g.
```
➜  cli-tools git:(verbose) ./crda analyse ~/works/log-parser/requirements.txt
❌ Please run `crda auth` command first.
```

^^ @prashbnair

**Comment by deepak1725** [comment]
> ## suggestions to auth command
> 
> ```
> To get Snyk Token, Please click https://app.snyk.io/redhat/snyk-token
> 
> Snyk Token [Press Enter to continue]: xxxx█
> Successfully Registered.
> 
> 	 crda-key: xxxx
> 
>  This token is confidential, Please keep it safe!
> ```
> 
> Seem to have some alignment issues.
> 
>     1. Leading space in `This token is confidential, Please keep it safe!` shall be removed
> 
>     2. Leading spaces in `	 crda-key: xxxx` shall be removed
> 
> 
> How about changing "Snyk Token [Press Enter to continue]" to "Snyk Token [Press Enter to skip]".?
> ## suggestion to analyse
> 
> ```
> Analysing your Dependency Stack! Please wait...
> Summary Report for Analyses:
> 
> Total Scanned Dependencies:68
> Total Scanned Transitive Dependencies:106
> Direct Vulnerable Dependencies:3
> Total Vulnerabilities:8
> Commonly Known Vulnerabilities:8
> Vulnerabilities Unique to Synk:0
> Critical Vulnerabilities:3
> High Vulnerabilities:0
> Medium Vulnerabilities:5
> Low Vulnerabilities:0
> 
> (Powered by Snyk)
> 
> Use --verbose for detailed report.%
> ```
> 
>     1. See a training % in my zsh, not sure it is the case for all
> 
>     2. Can you add space between : and number? e.g. `Total Scanned Dependencies: 68`
> 
>     3. Can this `Use --verbose for detailed report.` changed to `tip: use --verbose for detailed report.`
> 
> 
> ## suggestions to verbose
> 
> ```
> ➜  cli-tools git:(verbose) ./crda analyse ~/works/log-parser/requirements.txt --verbose
> Analysing your Dependency Stack! Please wait...
> Verbose Report for given Stack:
> 
> Scanned 68 Dependencies and 106 Transitives, Found 8 Issues
> 
> Fixable Issues:
> 
> 
> 
> 	 Issues in Transitives:
> 	 ✒ anymarkup@0.8.1->pyyaml@5.3.1
> 	 ✘ Arbitrary Code Execution[Critical Severity] [https://snyk.io/vuln/SNYK-PYTHON-PYYAML-590151] in pyyaml@5.3.1
> 	 introduced by anymarkup@0.8.1
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 	Upgrade cryptography@3.2.1 to cryptography@3.4.6
> 	 ✘ Cryptographic Issues[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-CRYPTOGRAPHY-1070544] in cryptography@3.2.1
> 	 introduced by cryptography@3.2.1
> 
> 
> 	 Issues in Transitives:
> 	 ✒ flask-cors@3.0.9->jinja2@2.11.2
> 	 ✘ Regular Expression Denial of Service (ReDoS)[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-JINJA2-1012994] in jinja2@2.11.2
> 	 introduced by flask-cors@3.0.9
> 
> 
> 
> 	 Issues in Transitives:
> 	 ✒ flask@1.1.2->jinja2@2.11.2
> 	 ✘ Regular Expression Denial of Service (ReDoS)[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-JINJA2-1012994] in jinja2@2.11.2
> 	 introduced by flask@1.1.2
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 	Upgrade jinja2@2.11.2 to jinja2@2.11.3
> 	 ✘ Regular Expression Denial of Service (ReDoS)[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-JINJA2-1012994] in jinja2@2.11.2
> 	 introduced by jinja2@2.11.2
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 	Upgrade pyyaml@5.3.1 to pyyaml@5.4.1
> 	 ✘ Arbitrary Code Execution[Critical Severity] [https://snyk.io/vuln/SNYK-PYTHON-PYYAML-590151] in pyyaml@5.3.1
> 	 introduced by pyyaml@5.3.1
> 
> 
> 
> 	 Issues in Transitives:
> 	 ✒ raven@6.10.0->jinja2@2.11.2
> 	 ✘ Regular Expression Denial of Service (ReDoS)[Medium Severity] [https://snyk.io/vuln/SNYK-PYTHON-JINJA2-1012994] in jinja2@2.11.2
> 	 introduced by raven@6.10.0
> 
> 
> 
> 
> 
> 
> 	 Issues in Transitives:
> 	 ✒ selinon@1.0.0->pyyaml@5.3.1
> 	 ✘ Arbitrary Code Execution[Critical Severity] [https://snyk.io/vuln/SNYK-PYTHON-PYYAML-590151] in pyyaml@5.3.1
> 	 introduced by selinon@1.0.0
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> (Powered by Snyk)
> ```
> 
>     1. See lots of empty spaces, not sure it is specific to my shell(zsh with oh-my-zsh)
> 
> 
> ^^ @deepak1725

Ack

**Comment by deepak1725** [comment]
> ```
> ➜  cli-tools git:(verbose) ./crda analyse ~/works/log-parser/requirements.txt
> 3:03PM FTL Please run `crda auth` command first to get `crda-key` and set it in environment.
> ```
> 
> I still see log statements with time and log level to show errors to cli users, IMO it should be replaced with normal error print(probably with emoji).
> 
> e.g.
> 
> ```
> ➜  cli-tools git:(verbose) ./crda analyse ~/works/log-parser/requirements.txt
> ❌ Please run `crda auth` command first.
> ```
> 
> ^^ @prashbnair

Ack.

**Review by arajkumar — APPROVED** [review]
lgtm,

wait for @prashbnair approval.

**Review by prashbnair — APPROVED** [review]

**Pull request merged**
